src/gt4py/next/program_processors/runners/dace_fieldview/transformations/map_orderer.py

```diff
@@ -0,0 +1,129 @@
+
+    Args:
+        leading_dim: A GT4Py dimension object that identifies the dimension that
+            is supposed to have stride 1.
+
+    Note:
+        The transformation does follow the rules outlines [here](https://hackmd.io/klvzLnzMR6GZBWtRU8HbDg#Requirements-on-SDFG)
+
+    Todo:
+        - Extend that different dimensions can be specified to be leading
+            dimensions, with some priority mechanism.
+        - Maybe also process the parameters to bring them in a canonical order.
+    """
+
+    leading_dim = properties.Property(
+        dtype=str,
+        allow_none=True,
+        desc="Dimension that should become the leading dimension.",
+    )
+
+    map_entry = transformation.transformation.PatternNode(nodes.MapEntry)
+
+    def __init__(
+        self,
+        leading_dim: Optional[Union[gtx_common.Dimension, str]] = None,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        if isinstance(leading_dim, gtx_common.Dimension):
+            self.leading_dim = dace_fieldview_util.get_map_variable(leading_dim)
+        elif leading_dim is not None:
+            self.leading_dim = leading_dim
+
+    @classmethod
+    def expressions(cls) -> Any:
+        return [dace.sdfg.utils.node_path_graph(cls.map_entry)]
+
+    def can_be_applied(
+        self,
+        graph: Union[SDFGState, SDFG],
+        expr_index: int,
+        sdfg: dace.SDFG,
+        permissive: bool = False,
+    ) -> bool:
+        """Test if the map can be reordered.
+
+        Essentially the function checks if the selected dimension is inside the map,
+        and if so, if it is on the right place.
+        """
+
+        if self.leading_dim is None:
+            return False
+        map_entry: nodes.MapEntry = self.map_entry
+        map_params: Sequence[str] = map_entry.map.params
+        map_var: str = self.leading_dim
+
+        if map_var not in map_params:
+            return False
+        if map_params[-1] == map_var:  # Already at the correct location
+            return False
+        return True
+
+    def apply(
+        self,
+        graph: Union[SDFGState, SDFG],
+        sdfg: SDFG,
+    ) -> None:
+        """Performs the actual parameter reordering.
+
+        The function will make the map variable, that corresponds to
+        `self.leading_dim` the last map variable (this is given by the structure of
+        DaCe's code generator).
+        """
+        map_entry: nodes.MapEntry = self.map_entry
+        map_params: list[str] = map_entry.map.params
+        map_var: str = self.leading_dim
+
+        # This implementation will just swap the variable that is currently the last
+        #  with the one that should be the last.
+        dst_idx = -1
+        src_idx = map_params.index(map_var)
+
+        for to_process in [
+            map_entry.map.params,
+            map_entry.map.range.ranges,
+            map_entry.map.range.tile_sizes,
```

Comment:
I was reviewing `map_orderer.py`. I do not have any comment on this file, however I've seen (I didn't know) that maps also have a `tile_sizes` property. Maybe it could be useful for K-blocking.

Reply:
This thing is actually deprecated and should be removed.
I have no idea what it does and when I asked the answer was essentially "do not use it, it is very old".